File: abc156d.cpp
#include<iostream>
using namespace std;

long long mod = 1000000000 + 7;
long long modinv_table[200000+1];


long long power(int x, int a){
   	if (a == 0)
		return 1;
	else if (a == 1)
		return x;
	else if (a % 2 == 0){
        long long te = power(x, a/2);
        return te * te % mod;
    }
    else{
        long long te = power(x, a/2);
        return (te * te)% mod * x % mod;
    }
}

long long modinv(int x){
	return power(x, mod-2);
}


long long binomial(int n, int k){
    long long ans = 1;
    for (int i=0; i<k; i++){
        ans *= n-i;
        ans %= mod;
        ans *= modinv_table[i+1];
        ans %= mod;
    }
    return ans;
}

int main()
{
    // スペース区切りの整数の入力
    int n, a, b;
    cin >> n >> a >> b;

    for (int i=1; i<b+1; i++){
        modinv_table[i] = modinv(i);
    }

    long long ans;
    ans = power(2, n) - 1;
    ans -= binomial(n, a);
    ans -= binomial(n, b);
    ans += (mod) * 3;

    ans %= mod;

    cout << ans << endl;
    return 0;
}

File: ABC_6q/abc156d.py
n, a, b = list(map(int, input().split()))

mod = 10**9 + 7

import time
start = time.time()

# xの逆元を求める。フェルマーの小定理より、 x の逆元は x ^ (mod - 2) に等しい。計算時間はO(log(mod))程度。
# https://qiita.com/Yaruki00/items/fd1fc269ff7fe40d09a6
def modinv(x):
	return pow(x, mod-2, mod)

modinv_table = [-1] * (b+1)
for i in range(1, b+1):
    modinv_table[i] = modinv(i)

end = time.time()
# print("time: ", end - start)

def binomial_coefficients2(n, k):
    ans = 1
    for i in range(k):
        ans *= n-i
        ans *= modinv_table[i + 1]
        ans %= mod
    return ans

# (2 ** n - 1) - nCa - nCb
ans = pow(2, n, mod) - 1

ans -= (binomial_coefficients2(n, a) + binomial_coefficients2(n, b))
print(ans % mod)


# print(binomial_coefficients2 (5, 2) % mod)
